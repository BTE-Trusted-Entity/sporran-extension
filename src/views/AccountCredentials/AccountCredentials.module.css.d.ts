declare namespace AccountCredentialsModuleCssNamespace {
  export interface IAccountCredentialsModuleCss {
    close: string;
    container: string;
    credentials: string;
    heading: string;
    subline: string;
    valid: string;
  }
}

declare const AccountCredentialsModuleCssModule: AccountCredentialsModuleCssNamespace.IAccountCredentialsModuleCss & {
  /** WARNING: Only available when `css-loader` is used without `style-loader` or `mini-css-extract-plugin` */
  locals: AccountCredentialsModuleCssNamespace.IAccountCredentialsModuleCss;
};

export = AccountCredentialsModuleCssModule;
