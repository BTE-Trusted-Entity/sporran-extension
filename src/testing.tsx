export * from '@testing-library/react';
export { waitForElementToBeRemoved } from '@testing-library/dom';
import { act, render as externalRender, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import dialogPolyfill from 'dialog-polyfill';

import { AccountsProviderMock } from './testing/AccountsProviderMock';

export {
  AccountsProviderMock,
  accountsMock,
} from './testing/AccountsProviderMock';

export { mockBackgroundScript } from './testing/mockBackgroundScript';

export function render(
  ui: Parameters<typeof externalRender>[0],
  options?: Parameters<typeof externalRender>[1],
): ReturnType<typeof externalRender> {
  return externalRender(
    <AccountsProviderMock>
      <MemoryRouter>{ui}</MemoryRouter>
    </AccountsProviderMock>,
    options,
  );
}

export async function waitForTooltipUpdate(): Promise<void> {
  expect(await screen.findByRole('status')).toHaveAttribute(
    'data-popper-escaped',
  );
}

const dialogPromise = Promise.resolve();
(dialogPolyfill.registerDialog as jest.Mock).mockReturnValue(dialogPromise);

export async function waitForDialogUpdate(): Promise<void> {
  await act(() => dialogPromise);
}
