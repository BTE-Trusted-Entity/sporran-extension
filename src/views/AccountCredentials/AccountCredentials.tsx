import { Link } from 'react-router-dom';
import { browser } from 'webextension-polyfill-ts';

import { Account } from '../../utilities/accounts/types';
import { AccountSlide } from '../../components/AccountSlide/AccountSlide';
import { Stats } from '../../components/Stats/Stats';
import { LinkBack } from '../../components/LinkBack/LinkBack';
import { paths } from '../paths';

import tableStyles from '../../components/Table/Table.module.css';
import styles from './AccountCredentials.module.css';

interface Props {
  account: Account;
}

export function AccountCredentials({ account }: Props): JSX.Element {
  const t = browser.i18n.getMessage;

  // TODO: use real data
  const credentials = [
    {
      Name: 'Email',
      'Full Name': 'Ingo RÃ¼be',
      Email: 'ingo@kilt.io',
      'Credential type': 'BL-Mail-Simple',
      Attester: 'BOTLabs',
      valid: true,
    },
  ];

  return (
    <section className={styles.container}>
      <h1 className={styles.heading}>{t('view_AccountCredentials_title')}</h1>
      <p className={styles.subline}>{t('view_AccountCredentials_subline')}</p>

      <AccountSlide account={account} />

      <table className={styles.credentials}>
        <thead>
          <tr className={tableStyles.tr}>
            <th className={tableStyles.th}>
              {t('view_AccountCredentials_name')}
            </th>
            <th className={tableStyles.th}>
              {t('view_AccountCredentials_ctype')}
            </th>
            <th className={tableStyles.th}>
              {t('view_AccountCredentials_attester')}
            </th>
            <th className={tableStyles.th}>
              {t('view_AccountCredentials_valid')}
            </th>
          </tr>
        </thead>
        <tbody>
          {credentials.map((credential) => (
            <tr key={credential.Name} className={tableStyles.tr}>
              <td className={tableStyles.td}>{credential.Name}</td>
              <td className={tableStyles.td}>
                {credential['Credential type']}
              </td>
              <td className={tableStyles.td}>{credential.Attester}</td>
              <td
                className={credential.valid ? styles.valid : tableStyles.td}
              />
            </tr>
          ))}
        </tbody>
      </table>

      <p>
        <Link to={paths.home} className={styles.close}>
          {t('common_action_close')}
        </Link>
      </p>

      <LinkBack />
      <Stats />
    </section>
  );
}
