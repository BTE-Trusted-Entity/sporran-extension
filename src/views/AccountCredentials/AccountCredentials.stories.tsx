import { Meta, Story } from '@storybook/react';

import { accountsMock } from '../../testing/AccountsProviderMock';

import { AccountCredentials } from './AccountCredentials';

type Type = Story<Parameters<typeof AccountCredentials>[0]>;

const account =
  accountsMock['4tJbxxKqYRv3gDvY66BKyKzZheHEH8a27VBiMfeGX2iQrire'];

export default {
  title: 'Views/AccountCredentials',
  component: AccountCredentials,
} as Meta;

export { AccountCredentials };
(AccountCredentials as Type).args = {
  account,
};
