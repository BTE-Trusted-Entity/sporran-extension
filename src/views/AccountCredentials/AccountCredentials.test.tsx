import { accountsMock, render } from '../../testing';

import { AccountCredentials } from './AccountCredentials';

const account =
  accountsMock['4tJbxxKqYRv3gDvY66BKyKzZheHEH8a27VBiMfeGX2iQrire'];

describe('AccountCredentials', () => {
  it('should render', async () => {
    const { container } = render(<AccountCredentials account={account} />);
    expect(container).toMatchSnapshot();
  });
});
