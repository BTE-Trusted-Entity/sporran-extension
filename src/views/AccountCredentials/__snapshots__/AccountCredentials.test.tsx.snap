// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountCredentials should render 1`] = `
<div>
  <section
    class="container"
  >
    <h1
      class="heading"
    >
      Your Credentials
    </h1>
    <p
      class="subline"
    >
      Overview
    </p>
    <section>
      <div
        class="MacFarlane tartan"
      >
        Identicon
      </div>
      <div
        class="name"
      >
        <span>
          My Sporran Account
        </span>
        <div
          class="wrapper"
        >
          <button
            aria-expanded="false"
            aria-haspopup="true"
            aria-label="Account options"
            class="toggle"
            role="button"
            tabindex="0"
            title="Account options"
            type="button"
          />
        </div>
      </div>
    </section>
    <table
      class="credentials"
    >
      <thead>
        <tr
          class="tr"
        >
          <th
            class="th"
          >
            Name
          </th>
          <th
            class="th"
          >
            C-Type
          </th>
          <th
            class="th"
          >
            Attester
          </th>
          <th
            class="th"
          >
            Valid
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class="tr"
        >
          <td
            class="td"
          >
            Email
          </td>
          <td
            class="td"
          >
            BL-Mail-Simple
          </td>
          <td
            class="td"
          >
            socialKYC
          </td>
          <td
            aria-label="Valid"
            class="valid"
          />
        </tr>
      </tbody>
    </table>
    <p>
      <a
        class="close"
        href="/"
      >
        Close
      </a>
    </p>
    <button
      aria-label="Back"
      class="linkBack"
      title="Back"
      type="button"
    />
  </section>
</div>
`;
